File: Level 0/코딩 기초 트레이닝/배열 만들기 6/solution.js
function solution(arr) {
  const stk = [];

  for (let i = 0; i < arr.length; i++) {
    if (!stk.length) {
      stk.push(arr[i]);
    } else if (stk.at(-1) === arr[i]) {
      stk.pop();
    } else if (stk.at(-1) !== arr[i]) {
      stk.push(arr[i]);
    }
  }

  return stk.length ? stk : [-1];
}

File: Level 0/코딩 기초 트레이닝/배열 만들기 6/solution.py
def solution(arr):
    stk = []

    for i in range(len(arr)):
        if not stk:
            stk.append(arr[i])
        elif stk and stk[-1] == arr[i]:
            stk.pop()
        elif stk and stk[-1] != arr[i]:
            stk.append(arr[i])

    return stk or [-1]
